patches/net/minecraft/server/level/ServerPlayer.java.patch

```diff
@@ -91,8 +91,31 @@
          return this.isReachableBedBlock(p_9117_) || this.isReachableBedBlock(p_9117_.relative(p_9118_.getOpposite()));
      }
  
-@@ -1003,6 +_,7 @@
-                     .send(new ClientboundOpenScreenPacket(abstractcontainermenu.containerId, abstractcontainermenu.getType(), p_9033_.getDisplayName()));
+@@ -983,6 +_,11 @@
+ 
+     @Override
+     public OptionalInt openMenu(@Nullable MenuProvider p_9033_) {
++        return openMenu(p_9033_, (java.util.function.Consumer<net.minecraft.network.FriendlyByteBuf>) null);
++    }
++
++    @Override
++    public OptionalInt openMenu(@Nullable MenuProvider p_9033_, @Nullable java.util.function.Consumer<net.minecraft.network.FriendlyByteBuf> extraDataWriter) {
```

Comment:
The vanilla logic will call `closeContainer`. Previously, the NetworkHooks method called `doCloseContainer` directly. There should be an option for modders to skip sending the `ClientboundContainerClosePacket` packet, else menu transitions in GUIs will needlessly re-center the cursor.